app/models/user.rb

```diff
@@ -1,10 +1,18 @@
 class User < ApplicationRecord
+  
+	before_destroy :keep_one_admin_user
 
-	has_many :tasks
+	has_many :tasks, dependent: :destroy
 	validates :name,  presence: true, length: { maximum: 30 }
-  validates :email, presence: true, length: { maximum: 255 },
-                    format: { with: /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i }
-  before_validation { email.downcase! }                  
+	validates :email, presence: true, length: { maximum: 255 },
+					format: { with: /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i }
+	before_validation { email.downcase! }                  
 	has_secure_password
 	
-end
+	def keep_one_admin_user
+		if User.where(admin: true).count == 1 && self.admin?
```

Comment:
好みもありますが、以下の記載も可能です。他にも.zero?, .any?, .many?などのメソッドも存在します。
```suggestion
		if User.where(admin: true).one? && self.admin?
```